test/bessel.jl

```diff
@@ -271,4 +271,54 @@ end
             @test f(0,1) ≈ f(0,Complex{Float16}(1))
         end
     end
+
+    @testset "jinc function" begin
+        for T in [Float16, Float32, Float64, Complex{Float16}, Complex{Float32},Complex{Float64}]
+            @test jinc(T(1.8)) ≈ T(-0.116401402395234261196060221416100814277048989838340270527643894)
+            @test jinc(T(0)) ≈ T(1)
+            x = 1.21966989126650445492653884746525517788
+            # check the first root for negative and positive
+            @test jinc(T(x)) + T(1) ≈ T(1)
+            @test jinc(T(-x)) + T(1) ≈ T(1)
+
+
+            # check for symmetry
+            for x = -5.0:1.0:3.0
+                @test jinc(T(10^x)) ≈ jinc(T(- 10^x))
+
+            end
+
+
+            x = T(1e-5)
+            res = T(0.999999999876629944991456407422485505370463)
+            @test jinc(x) ≈ res
+
+            x = T(2e-2)
+            res = T(0.9995066009475120760121935544369072906921634576)
+            @test jinc(x) ≈ res
+
+            # type stability for 0
+            @test typeof(jinc(T(0))) == T
```

Comment:
Make sure you test this for non-float types:
```jl
for T in (Int, Rational{Int}, BigInt,Float16,ComplexF16)
    @test @inferred(jinc(one(T))) === jinc(float(one(T)))
    @test jinc(zero(T)) === float(one(T)))
end
```
(`===` checks that the types match.)

For `Float16`, there should be a method that converts to `Float32` and then rounds back:
```jl
_jinc(x::Float16) = Float16(_jinc(Float32(x)))
_jinc(x::ComplexF16) = ComplexF16(_jinc(ComplexF32(x)))
```

Reply:
Ah, didn't know that.
For me that it feels like cheating to switch to higher precision in between? I assumed that we want to calculate the result using `Float16`. 
I'm not familiar with stuff like this...

Reply:
No, it is perfectly fine to switch to a higher-precision in between.   The only reason to use lower precision is if it is faster, but `Float16` is actually slower than `Float32` because only the latter is implemented in hardware on typical CPUs.

(`Float16` is mostly used as a storage format, or on GPUs.)